pkg/controller/job/job_controller_test.go

```diff
@@ -746,6 +761,13 @@ func TestControllerSyncJob(t *testing.T) {
 				// job & pods setup
 				job := newJob(tc.parallelism, tc.completions, tc.backoffLimit, tc.completionMode)
 				job.Spec.Suspend = pointer.Bool(tc.suspend)
+				if tc.presetJobStatus {
+					job.Status.Failed = tc.expectedFailed
```

Comment:
nit: base it on the input, rather than the expectations

```suggestion
					job.Status.Failed = tc.failedPods
```

Reply:
also count succeeded?

Reply:
`failedPods` creates failed pods with the tracking finalizer so the job ends up thinking that the failed pod has not been accounted for in the status and ends up incrementing `job.Status.Failed`. So in this case, pre-setting the status to `failedPods` does not work.

Reply:
ah, good point. In general it is beneficial for tests to be more verbose (following the DAMP principle). In this sense, we could actually have an explicit `initialStatus` object. But this recommendation is on the nit side.